internal/adapters/events/events_forwarder.go

```diff
@@ -217,12 +219,30 @@ func fromPlayerEventTypeToGrpcPlayerEventType(eventType events.PlayerEventType)
 	}
 }
 
-func handlerGrpcClientResponse(forwarder entities.Forwarder, eventResponse *pb.Response, err error) error {
+// handlerGrpcClientResponse checks if the gRPC response had unexpected communication or network errors.
+func handlerGrpcClientResponse(forwarder entities.Forwarder, eventResponse *pb.Response, err error) (codes.Code, error) {
 	if err != nil {
-		return err
+		grpcStatus, ok := status.FromError(err)
+		if !ok {
+			return codes.Unknown, errors.NewErrUnexpected("failed to forward event room at \"%s\" with unknown grpc code: %s", forwarder.Name, err.Error())
+		}
+
+		return grpcStatus.Code(), errors.NewErrUnexpected("failed to forward event room at \"%s\" with code %s", forwarder.Name, grpcStatus.Code().String())
 	}
-	if eventResponse.Code != 200 {
-		return errors.NewErrUnexpected("failed to forward event room at \"%s\"", forwarder.Name)
+
+	var ret codes.Code
+	// Matchmaker sends the status code as a http one, we normalize it to grpc codes for logs and metrics.
```

Comment:
nit: would replace with `Forwarder event's  "code" field is being used as an HTTP status code. we normalize it to grpc codes for logs and metrics`

This event definition is actually in this repo